public/scripts/tagsList.js

```diff
@@ -1,21 +1,148 @@
+    if (lists) {
+        lists.forEach(element => element.remove())
+    }
+}
+
+function setActive(autocompleteList) {
+    try {
+        setAllInactive()
+        autocompleteList.querySelectorAll('.unit-suggestion')[currentFocus].classList.add('autocomplete-active')
+    } catch (error) {
+        console.log(error)
+    }
 
 }
-(function(){
-    var tagInput = document.getElementById("tag-input")
-    var tagList = document.getElementById("tag-list")
-    tagInput.oninput =  () => {
-        // we will calculate the search options
+
+function setAllInactive() {
+    let autocompleteList = document.getElementById('autocomplete-list')
+    autocompleteList.querySelectorAll('.unit-suggestion').forEach(element => {
+        element.classList.remove('autocomplete-active');
+    })
+}
+
+function displayAutocompleteOptions(tagInput, suggestion) {
+    let template = document.createElement('template')
+    let searchValue = tagInput.value
+    let autocompleteDiv = document.createElement('div')
+    autocompleteDiv.setAttribute('class', 'autocomplete-list')
+    autocompleteDiv.setAttribute('id', 'autocomplete-list')
+
+    suggestion.forEach((element) => {
+        let unitSuggestion = document.createElement('div')
+        unitSuggestion.setAttribute('class', 'unit-suggestion')
+        unitSuggestion.innerHTML = `<strong>${element.substr(0, searchValue.length)}</strong>${element.substr(searchValue.length)} 
+        <input type='hidden' value="${element}">`
+
+        unitSuggestion.onclick = (e) => {
+            let currentInput = e.currentTarget.querySelector('input') // we might click on the <strong> tag so we need to bubble it up
+
+            if (currentInput) {
+                let value = e.currentTarget.querySelector('input').value
+                let tagList = document.getElementById('tag-list')
+                tagList.insertBefore(createTagObject(value), document.getElementById('autocomplete'))
+                tagInput.value = ""
+                closeRecommendationList()
+            }
+        }
+
+        autocompleteDiv.appendChild(unitSuggestion) // we stack them here
+    })
+
+    template.content.appendChild(autocompleteDiv)
+    tagInput.parentNode.appendChild(template.content)
+};
+
+
+
+(async function initializeLabelArray() {
+    try {
+        let response = await fetch('/api/labels', {
+            method: 'get',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            credentials: 'include'
+        })
+        labelArray = (await response.json()).content;
+    } catch (error) {
+        console.log(error)
     }
-    tagInput.onchange = () => {
-        tagList.insertBefore(getTagObject(tagInput.value),tagInput)
-        tagInput.value = ""
+})();
+
+(function setLabelTagEvents() {
+    var tagInput = document.getElementById('tag-input')
+    var tagList = document.getElementById('tag-list')
+    var autocomplete = document.getElementById('autocomplete')
```

Comment:
You can use `let`. I think, by default, you're pretty much safe to use `let`; that is, unless you need a `const`.

Reply:
It's the other way around: always use `const`, unless you need `let`!

Reply:
Brain fog 🙈